src/components/HelmRepositories/useRepositoryFormValidationSchema.tsx

```diff
@@ -0,0 +1,42 @@
+import { nameSchema } from '../../utils/commonValidationSchemas';
+import {
+  string as stringSchema,
+  boolean as booleanSchema,
+  object as objectSchema,
+  SchemaOf,
+} from 'yup';
+import { useTranslation } from '../../hooks/useTranslation';
+import { useArgoCDSecrets } from '../../hooks/useArgoCDSecrets';
+import { RepositoryFormValues } from './types';
+import React from 'react';
+
+const useRepositoryFormValidationSchema = (
+  isCreateFlow: boolean,
+): [SchemaOf<RepositoryFormValues>, boolean, unknown] => {
+  const { t } = useTranslation();
+  const [secrets, loaded, error] = useArgoCDSecrets();
+  const usedSecretNames = secrets.map((secret) => secret.data?.name);
```

Comment:
this line needs to be moved into the useMemo below, otherwise memoizing wont have any effect.
Also `useArgoCDSecrets` `decodedSecrets` needs to be memoized